js/indicators/chaikin.src.js

```diff
@@ -0,0 +1,141 @@
+    /**
+     * Chaikin Oscillator. This series requires the `linkedTo` option to
+     * be set and should be loaded after the `stock/indicators/indicators.js`
+     *
+     * @extends plotOptions.sma
+     * @product highstock
+     * @sample {highstock} stock/indicators/chaikin
+     *                     Chaikin Oscillator
+     * @since 7.0.0
+     * @excluding
+     *             allAreas,colorAxis,joinBy,keys,stacking,
+     *             showInNavigator,navigatorOptions,pointInterval,
+     *             pointIntervalUnit,pointPlacement,pointRange,pointStart
+     * @optionparent plotOptions.chaikin
+     */
+    {
+        /**
+         * Paramters used in calculation of Chaikin Oscillator
+         * series points.
+         * @excluding index
+         */
+        params: {
+            /**
+             * The id of volume series which is mandatory.
+             * For example using OHLC data, volumeSeriesID='volume' means
+             * the indicator will be calculated using OHLC and volume values.
+             *
+             * @type {String}
+             * @since 7.0.0
+             * @product highstock
+             */
+            volumeSeriesID: 'volume',
+            /**
+             * Periods for Chaikin Oscillator calculations.
+             * @default [3, 10]
+             * @type {Array}
+             * @since 7.0.0
+             */
+            periods: [3, 10]
+        }
+    }, /** @lends Highcharts.Series.prototype */ {
+        nameBase: 'Chaikin Osc',
+        nameComponents: ['periods'],
+        getValues: function (series, params) {
+            var periods = params.periods,
+                period = params.period,
+                ADL, // Accumulation Distribution Line data
+                CHA = [], // 0- date, 1- Chaikin Oscillator
+                xData = [],
+                yData = [],
+                periodsOffset,
+                SPE, // Shorter Period EMA
+                LPE, // Longer Period EMA
+                oscillator,
+                i;
+
+            // Check if periods are correct
+            if (periods.length !== 2 || periods[1] <= periods[0]) {
+                error(
+                    'Error: "Chaikin requires two periods. Notice, first ' +
+                    'period should be lower than the second one."'
+                );
+                return false;
+            }
+
+            ADL = AD.prototype.getValues.call(this, series, {
+                volumeSeriesID: params.volumeSeriesID,
+                period: period
+            });
+
+            SPE = EMA.prototype.getValues.call(this, ADL, {
+                period: periods[0]
+            });
+
+            LPE = EMA.prototype.getValues.call(this, ADL, {
+                period: periods[1]
+            });
+
+            // Check if ema is calculated properly, if not skip
+            if (!SPE || !LPE) {
+                return false;
+            }
+
+            periodsOffset = periods[1] - periods[0];
```

Comment:
How about situation when user set periods: [10,3] ?

Reply:
There is an if statement above that check it: _line 70_ `if (periods.length !== 2 || periods[1] <= periods[0])`.